prime-angular-frontend/src/app/modules/enrolment/pages/obo-sites-page/obo-sites-page.component.ts

```diff
@@ -259,16 +259,31 @@ export class OboSitesPageComponent extends BaseEnrolmentProfilePage implements O
    * allows for an empty list of oboSites if no jobs are solected.
    */
   private removeIncompleteOboSites(noEmptyOboSites: boolean = false) {
-    this.oboSites.controls
-      .forEach((control: FormGroup, index: number) => {
-        const value = control.get('physicalAddress').value.city;
-        const careSetting = control.get('careSettingCode').value;
-
-        // Remove when empty, default option, or group is invalid
-        if (!value || value === this.defaultOptionLabel || control.invalid) {
-          this.removeOboSite(index, careSetting);
+    for (const [key, value] of Object.entries(this.enrolmentFormStateService.oboSitesForm.controls)) {
```

Comment:
Let's have a chat about how you think this form works on and off the page, and what you need to do to get the result you need without making this kind of change.